test/framework/src/main/java/org/elasticsearch/test/disruption/DisruptableMockTransport.java

```diff
@@ -0,0 +1,188 @@
+                            }
+
+                            @Override
+                            public String getChannelType() {
+                                return "test-channel";
+                            }
+
+                            @Override
+                            public void sendResponse(final TransportResponse response) {
+                                sendFromTo(destination, getLocalNode(), new Runnable() {
+                                    @Override
+                                    public void run() {
+                                        if (getConnectionStatus(destination, getLocalNode()) != ConnectionStatus.CONNECTED) {
+                                            logger.trace("dropping response to {}: channel is not CONNECTED",
+                                                requestDescription);
+                                        } else {
+                                            handleResponse(requestId, response);
+                                        }
+                                    }
+
+                                    @Override
+                                    public String toString() {
+                                        return "response to " + requestDescription;
+                                    }
+                                });
+                            }
+
+                            @Override
+                            public void sendResponse(TransportResponse response,
+                                                     TransportResponseOptions options) {
+                                sendResponse(response);
+                            }
+
+                            @Override
+                            public void sendResponse(Exception exception) {
+                                sendFromTo(destination, getLocalNode(), new Runnable() {
+                                    @Override
+                                    public void run() {
+                                        if (getConnectionStatus(destination, getLocalNode()) != ConnectionStatus.CONNECTED) {
+                                            logger.trace("dropping response to {}: channel is not CONNECTED",
+                                                requestDescription);
+                                        } else {
+                                            handleRemoteError(requestId, exception);
+                                        }
+                                    }
+
+                                    @Override
+                                    public String toString() {
+                                        return "error response to " + requestDescription;
+                                    }
+                                });
+                            }
+                        };
+
+                        try {
+                            requestHandler.processMessageReceived(request, transportChannel);
```

Comment:
IntelliJ complains about lack of genericness for this call. Could it move to its own method on which such warnings are suppressed?